docs/architecture/adr-022-per-subspace-token-factory.md

```diff
@@ -0,0 +1,228 @@
+# ADR 022: Per-subspace token factory
+
+## Changelog
+- April 28th, 2023: First draft;
+
+## Status
+
+Draft
+
+## Abstract
+
+This ADR introduces a new feature that enables the subspace admins to create, mint and burn a subspace related
+coin. 
+
+## Context
+
+Desmos is a social network protocol that allows users to create, share, and engage with content 
+on a decentralized platform. 
+Currently, to post or interact with the content in a subspace you need to pay the gas fee using DSM. 
+This have the consequence of not giving any financial incentive on creating a social network on Desmos.
+
+## Decision
+
+We will wrap the Osmosis [Token Factory module](https://docs.osmosis.zone/osmosis-core/modules/tokenfactory/)
```

Comment:
Should we use the (hopefully more stable) CosmWasm module instead as @dadamu linked [here](https://github.com/desmos-labs/desmos/pull/1128/files#r1184576805)?

Reply:
Yes I think it's better if we use the CosmWasm module since don't depends on a forked cosmos sdk with a modified bank module.